src/main/java/Domain/Product.java

```diff
@@ -99,6 +73,11 @@ public synchronized void cancelPurchase() {
                 + " had been purchased cancel -- +1 to stock.");
     }
 
+    // this function add a review to the product
+    public void addReview(String username, String review) {
```

Comment:
amazing, consider use of concrerate map or  something like that, Idk if it needed tho